qds_sdk/commands.py

```diff
@@ -1316,7 +1316,18 @@ def _is_complete_data_available(bucket_paths, num_result_dir):
             raise Exception("Results file not available on s3 yet. This can be because of s3 eventual consistency issues.")
         log.info("Downloading file from %s" % s3_path)
         if delim is None:
-            key_instance.get_contents_to_file(fp)  # cb=_callback
+            try:
+                key_instance.get_contents_to_file(fp)  # cb=_callback
+            except boto.exception.S3ResponseError as e:
+                if (e.status == 403):
+                    # SDK-191, boto gives an error while fetching the objects using versions which happens by default
+                    # in the get_contents_to_file() api. So attempt one without specifying version.
+                    log.info("Access denied while fetching the s3 object. Retrying without specifying the version....")
```

Comment:
Also have we checked with AWS (or read in their sdk docs) of this behaviour ? This seems like a hack. Are we able to reproduce it in house ?